pkg/store/configuration/watchers.go

```diff
@@ -0,0 +1,69 @@
+// SPDX-FileCopyrightText: ${year}-present Open Networking Foundation <info@opennetworking.org>
+// SPDX-License-Identifier: LicenseRef-ONF-Member-Only-1.0
+
+package configuration
+
+import (
+	"sync"
+
+	"github.com/google/uuid"
+	configapi "github.com/onosproject/onos-api/go/onos/config/v2"
+)
+
+type watcher struct {
+	eventCh chan<- configapi.ConfigurationEvent
+	replay  bool
+	watchID configapi.ConfigurationID
+}
+
+// watchers stores the information about watchers
+type watchers struct {
+	watchers map[uuid.UUID]watcher
+	rm       sync.RWMutex
+}
+
+// newWatchers creates watchers
+func newWatchers() *watchers {
+	return &watchers{
+		watchers: make(map[uuid.UUID]watcher),
+	}
+}
+
+// sendAll sends a configuration event for all registered watchers
+func (ws *watchers) sendAll(event configapi.ConfigurationEvent) {
+	ws.rm.RLock()
```

Comment:
This read lock will have no effect because the actual read is being don’t in a different goroutine. It will be released before the memory is even read inside the new goroutine. The lock needs to be acquired by the goroutine that’s reading the memory.